src/riak_dt_orswot.erl

```diff
@@ -260,26 +273,34 @@ remove_all([Elem | Rest], Actor, ORSet, Ctx) ->
     {ok, ORSet2} =  update({remove, Elem}, Actor, ORSet, Ctx),
     remove_all(Rest, Actor, ORSet2, Ctx).
 
--spec merge(orswot(), orswot()) -> orswot().
+-spec merge(orswot() | legacy_orswot(), orswot() | legacy_orswot()) -> orswot().
+merge({LHSClock, LHSEntries0, LHSDeferred}, RHS) when is_list(LHSEntries0) ->
+    LHSEntries1 = maps:from_list(LHSEntries0),
+    merge({LHSClock, LHSEntries1, LHSDeferred}, RHS);
+merge(LHS, {RHSClock, RHSEntries0, RHSDeferred}) when is_list(RHSEntries0) ->
+    RHSEntries1 = maps:from_list(RHSEntries0),
+    merge(LHS, {RHSClock, RHSEntries1, RHSDeferred});
 merge({Clock, Entries, Deferred}, {Clock, Entries, Deferred}) ->
     {Clock, Entries, Deferred};
+merge({Clock, Entries, LHSDeferred}, {Clock, Entries, RHSDeferred}) ->
+    Deffered = merge_deferred(LHSDeferred, RHSDeferred),
+    apply_deferred(Clock, Entries, Deffered);
+merge(?EMPTY_ORSWOT, RHS) -> RHS;
+merge(LHS, ?EMPTY_ORSWOT) -> LHS;
```

Comment:
You want to do that only if it is non-empty. For Empty ones there is no merge, right?